File: app/src/main/java/com/sup/dev/android/views/widgets/pager/ViewPagerIndicatorTitles.java
package com.sup.dev.android.views.widgets.pager;

import android.content.Context;
import android.content.res.TypedArray;
import android.support.annotation.MainThread;
import android.util.AttributeSet;
import android.view.View;
import android.widget.TextView;

import com.sup.dev.android.androiddevsup.R;
import com.sup.dev.android.app.SupAndroid;
import com.sup.dev.android.utils.interfaces.UtilsView;

public class ViewPagerIndicatorTitles extends ViewPagerIndicator implements View.OnClickListener {

    private final UtilsView utilsView;
    private int resId;
    private int titleId = R.layout.w_indicator_title;
    private int offsetLeft = R.id.title;

    private TextView[] views;


    @MainThread
    public ViewPagerIndicatorTitles(Context context, AttributeSet attrs) {
        super(context, attrs);

        SupAndroid.initEditMode(this);
        utilsView = SupAndroid.di.utilsView();

        TypedArray a = context.obtainStyledAttributes(attrs, R.styleable.ViewPagerIndicatorTitles, 0, 0);
        offsetLeft = (int) a.getDimension(R.styleable.ViewPagerIndicatorTitles_ViewPagerIndicatorTitles_offset_left, offsetLeft);
        resId = a.getResourceId(R.styleable.ViewPagerIndicatorTitles_ViewPagerIndicatorTitles_titleLayout, resId);
        titleId = a.getResourceId(R.styleable.ViewPagerIndicatorTitles_ViewPagerIndicatorTitles_titleId, titleId);
        a.recycle();
    }

    public void reset() {
        //removeAllViews();
        //views = new TextView[pager.getPageCount()];
        //for (int i = 0; i < views.length; i++) {
        //    views[i] = utilsView.inflate(getContext(), resId);
        //    views[i].setText(pager.getPageTitle(i));
        //    views[i].setOnClickListener(this);
        //    addView(views[i]);
        //}
    }

    @Override
    @MainThread
    protected void onChanged() {
        requestLayout();
    }

    @Override
    public void onClick(View v) {
        //  for (int i = 0; i < views.length; i++)
        //      if (v == views[i])
        //          pager.toPage(i);
    }

    @Override
    @MainThread
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        for (int i = 0; i < getChildCount(); i++) {
            View v = getChildAt(i);
            measureChild(v, widthMeasureSpec, heightMeasureSpec);
        }
        setMeasuredDimension(widthMeasureSpec, heightMeasureSpec);
    }

    @Override
    @MainThread
    protected void onLayout(boolean changed, int l, int t, int r, int b) {
        super.onLayout(changed, l, t, r, b);
        final int count = getChildCount();
        if (count == 0) return;
        int xOffset = 0;
        for (int i = 0; i < position; i++) {
            View child = getChildAt(i);
            xOffset += child.getMeasuredWidth() + offset;
        }

        // TextView selected = (views[position].findViewById(titleId));
        // TextView old = null;
        // int oldIndex = (positionOffset > 0) ? position + 1 : position - 1;
        // if (positionOffset != 0 && oldIndex > -1 && oldIndex < count) {
        //     old = (views[oldIndex].findViewById(titleId));
        //     xOffset += ((selected.getMeasuredWidth() + old.getMeasuredWidth()) / 2 + offset) * positionOffset;
        // }

        // int x = (getWidth() - offsetLeft - selected.getMeasuredWidth()) / 2 - xOffset;
        // for (int i = 0; i < count; i++) {
        //     View child = getChildAt(i);
        //     int y = (getHeight() - child.getMeasuredHeight()) / 2;
        //     child.layout(x, y, x + child.getMeasuredWidth(), child.getMeasuredHeight() + y);
        //     x += offset + child.getMeasuredWidth();

        //     TextView textView = (views[i].findViewById(titleId));
        //     textView.setTextColor(ToolsColor.setAlpha(textView.getCurrentTextColor(), 120));
        // }

        //  if (positionOffset != 0) {
        //      float arg = Math.abs(positionOffset);
        //      selected.setTextColor(ToolsColor.setAlpha(selected.getCurrentTextColor(), 120 + (int) ((255 - 120) * (1 - arg))));
        //      old.setTextColor(ToolsColor.setAlpha(old.getCurrentTextColor(), 120 + (int) ((255 - 120) * arg)));
        //  } else {
        //      selected.setTextColor(ToolsColor.setAlpha(selected.getCurrentTextColor(), 255));
        //  }


    }


}

File: src/main/java/com/sup/dev/android/views/views/pager/ViewPagerIndicatorTitles.kt
package com.sup.dev.android.views.views.pager

import android.content.Context
import androidx.annotation.MainThread
import android.util.AttributeSet
import android.view.View
import android.widget.TextView
import com.sup.dev.android.R
import com.sup.dev.android.tools.ToolsResources
import com.sup.dev.android.tools.ToolsView
import com.sup.dev.java.tools.ToolsColor

class ViewPagerIndicatorTitles @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null) : ViewPagerIndicatorViews(context, attrs) {


    private var titles: Array<out String?>? = null

    fun setTitles(vararg titles: String) {
        this.titles = titles
    }

    fun setTitles(vararg titles: Int) {
        this.titles = arrayOfNulls(titles.size)
        for (i in titles.indices)
            (this.titles as Array<String?>)[i] = ToolsResources.s(titles[i])
    }

    override fun instanceView(index: Int): View {
        val v:TextView = ToolsView.inflate(context, R.layout.view_indicator_title)
        v.text = if (titles!!.size <= index) null else titles!![index]
        return v
    }

    @MainThread
    override fun onLayout(changed: Boolean, l: Int, t: Int, r: Int, b: Int) {
        super.onLayout(changed, l, t, r, b)
        val count = childCount
        if (count == 0) return

        val selected = views[position].findViewById<TextView>(R.id.vDevSupTitle)
        var old: TextView? = null
        val oldIndex = if (positionOffset > 0) position + 1 else position - 1
        if (positionOffset != 0f && oldIndex > -1 && oldIndex < count) {
            old = views[oldIndex].findViewById(R.id.vDevSupTitle)
        }

        for (i in 0 until count) {
            val textView = views[i].findViewById<TextView>(R.id.vDevSupTitle)
            textView.setTextColor(ToolsColor.setAlpha(120, textView.currentTextColor))
        }

        if (positionOffset != 0f) {
            val arg = Math.abs(positionOffset)
            selected.setTextColor(ToolsColor.setAlpha(120 + ((255 - 120) * (1 - arg)).toInt(), selected.currentTextColor))
            old?.setTextColor(ToolsColor.setAlpha(120 + ((255 - 120) * arg).toInt(), old.currentTextColor))
        } else {
            selected.setTextColor(ToolsColor.setAlpha(255, selected.currentTextColor))
        }


    }

}